src/composables/useTemplateFiltering.ts

```diff
@@ -1,57 +1,224 @@
+import Fuse from 'fuse.js'
 import { type Ref, computed, ref } from 'vue'
 
 import type { TemplateInfo } from '@/types/workflowTemplateTypes'
 
 // @ts-expect-error unused (To be used later?)
 interface TemplateFilterOptions {
   searchQuery?: string
+  selectedModels?: string[]
+  selectedUseCases?: string[] // Now represents selected tags
+  selectedLicenses?: string[]
+  sortBy?:
+    | 'default'
+    | 'alphabetical'
+    | 'newest'
+    | 'vram-low-to-high'
+    | 'model-size-low-to-high'
 }
 
 export function useTemplateFiltering(
   templates: Ref<TemplateInfo[]> | TemplateInfo[]
 ) {
   const searchQuery = ref('')
+  const selectedModels = ref<string[]>([])
+  const selectedUseCases = ref<string[]>([])
+  const selectedLicenses = ref<string[]>([])
```

Comment:
Not for this PR, but at some point can we change these from `string` to a string union enum?